internal/handlers/gateway/modes.go

```diff
@@ -0,0 +1,31 @@
+package gateway
+
+import (
+	"fmt"
+
+	"github.com/os-observability/tempo-operator/apis/tempo/v1alpha1"
+)
+
+// ValidateModes validates the tenants mode specification.
+func ValidateModes(tempo v1alpha1.TempoStack) error {
+	if tempo.Spec.Tenants == nil {
+		return nil
+	}
+
+	if tempo.Spec.Tenants.Mode == v1alpha1.Static {
+		if tempo.Spec.Tenants.Authentication == nil {
+			return fmt.Errorf("mandatory configuration - missing tenants' authentication configuration")
+		}
+
+		if tempo.Spec.Tenants.Authorization == nil || tempo.Spec.Tenants.Authorization.Roles == nil {
+			return fmt.Errorf("mandatory configuration - missing roles configuration")
+		}
+
+		if tempo.Spec.Tenants.Authorization == nil || tempo.Spec.Tenants.Authorization.RoleBindings == nil {
+			return fmt.Errorf("mandatory configuration - missing role bindings configuration")
+		}
+	}
+
+	// TODO: support openshift mode.
```

Comment:
Could you create an issue and place the ID here? Or maybe remove this and just create the ticket

Reply:
If you create a ticket please add a link to the source code as well